services/sources/publisher_backlist_service.py

```diff
@@ -23,6 +24,48 @@ def __init__(self):
         
         self.airtable_auth_token = os.environ.get('AIRTABLE_KEY', None)
 
+    def delete_records(
+        self,
+        offset: Optional[int]=None,
+        limit: Optional[int]=None
+    ):
+        filter_by_formula = self.build_filter_by_formula_parameter(deleted=True, full_import=False, start_timestamp=None)
+        
+        array_json_records = self.get_records_array(limit, filter_by_formula)
+
+        for json_dict in array_json_records:
+            if json_dict['records']:
+                for records_value in json_dict['records']:
+                    try:
+                        record_metadata_dict = records_value['fields']
+                        pub_backlist_record = PublisherBacklistMapping(record_metadata_dict)
+                        pub_backlist_record.applyMapping()
+                        self.delete_manifest(record_metadata_dict, pub_backlist_record.record)
+                        self.delete_digital_asset_file()
+                    except Exception:
+                        logger.exception(f'Record has not been created {records_value}')
+        
+    def delete_manifest(self, record_metadata_dict, mapped_record):
+        try:
+            for record in self.session.query(Record) \
+                .filter(Record.source_id == mapped_record.source_id):  
+                    key_format = f"{self.prefix}{record.source}"
+
+                    if record_metadata_dict['File ID 1']:
+                        file_title = record_metadata_dict['File ID 1']
```

Comment:
Hmm is there an easier way to do get the manifest file name? Is this anywhere in the record?

Reply:
The only alternative I could think of is using the record.has_part attribute, splitting the list element that contains the link to the webpub manifest (application/webpub+json type) and splicing the link string to retrieve the file id from that link.